code/game/objects/structures/fence.dm

```diff
@@ -0,0 +1,246 @@
+#define SMALL_HOLE 1 //small hole in the section - can pass small items through.
+#define MEDIUM_HOLE 2 //medium hole in the section - can climb through (takes 20 seconds)
+#define LARGE_HOLE 3 //large hole in the section - can walk through
+#define MAX_HOLE_SIZE LARGE_HOLE
+
+/obj/structure/fence
+	name = "fence"
+	desc = "A chain link fence. Not as effective as a wall, but generally it keeps people out."
+	density = 1
+	anchored = 1
+
+	icon = 'icons/obj/structures/fence.dmi'
+	icon_state = "straight"
+
+	var/cuttable = TRUE
+	var/hole_size= NO_HOLE
+	var/invulnerable = FALSE
+
+/obj/structure/fence/New()
+	..()
+
+	update_cut_status()
+
+/obj/structure/fence/examine(mob/user)
+	.=..()
+
+	switch(hole_size)
+		if(SMALL_HOLE)
+			user.show_message("There is a small hole in \the [src].", MESSAGE_SEE)
+		if(MEDIUM_HOLE)
+			user.show_message("There is a large hole in \the [src].", MESSAGE_SEE)
+		if(LARGE_HOLE)
+			user.show_message("\The [src] has been completely cut through.", MESSAGE_SEE)
+
+/obj/structure/fence/end
+	icon_state = "end"
+	cuttable = FALSE
+
+/obj/structure/fence/corner
+	icon_state = "corner"
+	cuttable = FALSE
+
+/obj/structure/fence/post
+	icon_state = "post"
+	cuttable = FALSE
+
+/obj/structure/fence/cut/small
+	icon_state = "straight_cut1"
+	hole_size = SMALL_HOLE
+
+/obj/structure/fence/cut/medium
+	icon_state = "straight_cut2"
+	hole_size = MEDIUM_HOLE
+
+/obj/structure/fence/cut/large
+	icon_state = "straight_cut3"
+	hole_size = LARGE_HOLE
+
+/obj/structure/fence/attackby(obj/item/W, mob/user)
+	if(iswirecutter(W))
+		if(!cuttable)
+			to_chat(user, "<span class='notice'>This section of the fence can't be cut.</span>")
+			return
+
+		if(invulnerable)
+			to_chat(user, "<span class='notice'>This fence is too strong to cut through.</span>")
+			return
+
+		var/current_stage = hole_size
+		if(current_stage >= MAX_HOLE_SIZE)
+			return
+
+		user.visible_message("<span class='danger'>\The [user] starts cutting through \the [src] with \the [W].</span>",\
+		"<span class='danger'>You start cutting through \the [src] with \the [W].</span>")
+
+		if(do_after(user, src, CUT_TIME))
+			if(current_stage == hole_size)
+
+				switch(++hole_size)
+					if(SMALL_HOLE)
+						visible_message("<span class='notice'>\The [user] creates a small opening in \the [src] with \the [W].</span>")
+						to_chat(user, "<span class='info'>This hole seems to be [user.is_fat() ? "way " : ""]too small to climb though, but you probably could throw something through it.</span>")
+					if(MEDIUM_HOLE)
+						visible_message("<span class='notice'>\The [user] cuts into \the [src] some more.</span>")
+						if(user.is_fat())
+							to_chat(user, "<span class='info'>While a thinner person could climb through this hole, it's still too small for you.</span>")
+						else
+							to_chat(user, "<span class='info'>You could probably fit yourself through that hole now. Although it would be better if you made it slightly bigger.</span>")
```

Comment:
20 seconds seems a bit long for climbing through
And Celt, it takes 20 seconds to climb through *each time*